app/controllers/api/v2/hosts_controller.rb

```diff
@@ -126,6 +126,30 @@ def status
         render :json => { :status => @host.host_status }.to_json if @host
       end
 
+      api :GET, "/hosts/:id/vm_compute_attributes", N_("Get vm attributes of host")
+      param :id, :identifier_dottable, :required => true
+      description <<-eos
+Return the host's compute attributes that can be used to create a clone of this VM
+      eos
+
+      def vm_compute_attributes
+        if @host
```

Comment:
If the whole method is going to be wrapped in this if clause, you can use a guard clause `render :json => {} unless @host.present?` and put everything else in the method (without the if)